src/sensai/geoanalytics/graph.py

```diff
@@ -1,11 +1,11 @@
-import geopandas as gp
```

Comment:
This is a bit unfortunate since graph stuff is not necessarily related to coordinates, only the coordinate spanning tree does. You could move the delaunayGraph and SpanningTree back to util.graph and remove the
`extractCoordinatesArray(datapoints)` in the latter, to decouple it from geopandas

Reply:
It is related to coordinates, since a Delaunay triangulation only makes sense for euclidean points. If we would like to have stuff only related to the graph representation, it shouldn't be tightly coupled to a special type of graph (representing a Delaunay triangulation). The class 

`class SpanningTree:
    """
    Wrapper around a tree-finding algorithm that will be applied on the Delaunay graph of the datapoints
    """
    def __init__(self, datapoints: np.ndarray, tree_finder: Callable[[nx.Graph], nx.Graph] = nx.minimum_spanning_tree):`


should not depend on an np.ndarray, which has to be an array of euclidean coordinates, since a Delaunay triangualation is computed, but rather a nx.Graph. Moreover, paramter tree_finder allows to construct a subgraph, which may not be a spanning tree at all. Actually, the class only provides convenience methods for inspecting a weighted (nx) graph. So, either the class should be

`class WeightedGraphWrapper:
    """
    Wrapper around a nx.Graph
    """
    def __init__(self, graph: nx.Graph):`

Or, if you really would like to have a spanning tree, it may look like

`class SpanningTree:
    def __init__(self, graph: nx.Graph, mode="min"):
        self.tree = nx.minimum_spanning_tree if mode== "min" else ...`

(resp. including an enum for min max). In any case, I don't know if this is enough functionality, to justify a class.
By now it is tightly coupled to euclidean point graphs (even more, Delaunay graphs), so I would put it into geoanalytics. What do you think?

Reply:
I agree that the name SpanningTree suggests more generality than really is there. Let us keep it as is in your PR for now, I will open a separate issue for generalizing it (which might never get done if not needed, but at least the problem will be documented). 

The only thing I disagree with is that because it is based on numpy arrays it belongs to geospatial - it would work for anything Euclidian, whereas geospatial is typically restricted to 2 or 3 dim. But since it is not general enough to deserve a place in util/graph and I don't want to fix that now, let's move it :). Feel free to resolve the thread